File: app/src/main/java/ch/unstable/ost/SectionDetailFragment.java
package ch.unstable.ost;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.Arrays;

import ch.unstable.ost.api.model.Section;
import ch.unstable.ost.views.lists.station.SectionsStopsListAdapter;

public class SectionDetailFragment extends Fragment {

    private static final String KEY_SECTION = "SectionDetailFragment.KEY_SECTION";
    private Section mSection;
    private SectionsStopsListAdapter mStopsListAdapter;

    public static SectionDetailFragment newInstance(Section section) {
        Bundle arguments = new Bundle();
        arguments.putParcelable(KEY_SECTION, section);
        SectionDetailFragment fragment = new SectionDetailFragment();
        fragment.setArguments(arguments);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (savedInstanceState != null) {
            mSection = savedInstanceState.getParcelable(KEY_SECTION);
        } else {
            mSection = getArguments().getParcelable(KEY_SECTION);
        }
        if (mSection == null) {
            throw new IllegalStateException("section not set");
        }
        mStopsListAdapter = new SectionsStopsListAdapter();
        mStopsListAdapter.setElements(mSection.getStops());
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putParcelable(KEY_SECTION, mSection);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_journey_detail, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        RecyclerView stationsList = view.findViewById(R.id.stationsList);
        stationsList.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
        stationsList.setAdapter(mStopsListAdapter);
    }
}

File: app/src/main/java/ch/unstable/ost/SectionDetailFragment.kt
package ch.unstable.ost

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import ch.unstable.ost.api.model.Section
import ch.unstable.ost.views.lists.station.SectionsStopsListAdapter

class SectionDetailFragment : Fragment() {
    private var sectionToShow: Section? = null
    private var stopsListAdapter: SectionsStopsListAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sectionToShow = if (savedInstanceState != null) {
            savedInstanceState.getParcelable(KEY_SECTION)
        } else {
            requireArguments().getParcelable(KEY_SECTION)
        }
        checkNotNull(sectionToShow) { "section not set" }
        stopsListAdapter = SectionsStopsListAdapter().apply {
            // TODO
            setElements(emptyList())
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putParcelable(KEY_SECTION, sectionToShow)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_journey_detail, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val stationsList: RecyclerView = view.findViewById(R.id.stationsList)
        stationsList.layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
        stationsList.adapter = stopsListAdapter
    }

    companion object {
        private const val KEY_SECTION = "SectionDetailFragment.KEY_SECTION"
        fun newInstance(section: Section?): SectionDetailFragment {
            val arguments = Bundle()
            arguments.putParcelable(KEY_SECTION, section)
            val fragment = SectionDetailFragment()
            fragment.arguments = arguments
            return fragment
        }
    }
}